ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -102,7 +103,13 @@ private static LogicalSchema buildSchema(final PlanNode left, final PlanNode rig
     for (final Field field : rightSchema.valueSchema().fields()) {
       schemaBuilder.field(field.name(), field.schema());
     }
-    return LogicalSchema.of(schemaBuilder.build());
+
+    final ConnectSchema keySchema = left.getSchema().withoutAlias().keySchema();
```

Comment:
The key for the join may not be the same as the key for the left hand side if the left hand side key is not the same as the join key. KSQL will repartition the left side stream with a new key and this key will be the key for the result of the join.

Reply:
You're right, this isn't correct, though works at the moment as its always ROWKEY/STRING. I'll change to hardcode to ROWKEY/STRING for now.  This will be changed in later PRs are the type / name of keys can be other than ROWKEY.